pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -628,19 +628,28 @@ func lookUpContainerInPod(containerID string, status corev1.PodStatus) (*corev1.
 	return nil, containerNotInPod
 }
 
-func getPodImages(containerStatusArray []corev1.ContainerStatus) map[string]bool {
+func getPodImageIdentifiers(containerStatusArray []corev1.ContainerStatus) map[string]bool {
+	// Map is used purely to exclude duplicate selectors, value is unused.
 	podImages := make(map[string]bool)
-
-	// collect container images
+	// Note that for each pod image we generate *2* matching selectors.
+	// This is to support matching against ImageID, which has a SHA
+	// docker.io/envoyproxy/envoy-alpine@sha256:bf862e5f5eca0a73e7e538224578c5cf867ce2be91b5eaed22afc153c00363eb
+	// as well as
+	// docker.io/envoyproxy/envoy-alpine:v1.16.0, which does not,
+	// while also maintaining backwards compatibility and allowing for dynamic attestation
+	// when the SHA is not yet known (e.g. before the image pull is initiated)
```

Comment:
> when the SHA is not yet known
Does this mean that status.ImageID could be blank? Should we be checking for that before adding to the map?

Reply:
@blegget/@caleblloyd, any insight into my question? If either of those fields can be empty, we should avoid polluting the map, but I'm not sure at all if that is possible.

Reply:
I am not that familiar with the Kubelet API - but I think that it would be worth adding an empty-check to `status.Image` and `status.ImageID` to be safe